### ISSUE TITLE
Tests failure with latest release

### ISSUE THREAD

**Issue by ArchangeGabriel** [mwilliamson/locket.py#13]
This is on Arch Linux :
```
======================================================================
ERROR: tests.locket_tests.can_set_timeout_to_zero_to_raise_exception_if_lock_cannot_be_acquired
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/lib/python3.9/site-packages/nose/case.py", line 197, in runTest
    self.test(*self.arg)
  File "/build/python-locket/src/locket.py-0.2.1/tests/locket_tests.py", line 25, in run_test
    return func(lock_path)
  File "/build/python-locket/src/locket.py-0.2.1/tests/locket_tests.py", line 196, in can_set_timeout_to_zero_to_raise_exception_if_lock_cannot_be_acquired
    locker_2.acquire()
  File "/build/python-locket/src/locket.py-0.2.1/tests/locket_tests.py", line 274, in acquire
    self._process.stdin_write(b"\n")
  File "/usr/lib/python3.9/site-packages/spur/local.py", line 178, in stdin_write
    self._process_stdin.write(value)
BrokenPipeError: [Errno 32] Broken pipe

======================================================================
ERROR: tests.locket_tests.error_is_raised_after_timeout_has_expired
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/lib/python3.9/site-packages/nose/case.py", line 197, in runTest
    self.test(*self.arg)
  File "/build/python-locket/src/locket.py-0.2.1/tests/locket_tests.py", line 25, in run_test
    return func(lock_path)
  File "/build/python-locket/src/locket.py-0.2.1/tests/locket_tests.py", line 220, in error_is_raised_after_timeout_has_expired
    locker_2.acquire()
  File "/build/python-locket/src/locket.py-0.2.1/tests/locket_tests.py", line 274, in acquire
    self._process.stdin_write(b"\n")
  File "/usr/lib/python3.9/site-packages/spur/local.py", line 178, in stdin_write
    self._process_stdin.write(value)
BrokenPipeError: [Errno 32] Broken pipe

======================================================================
ERROR: tests.locket_tests.lock_file_blocks_until_lock_is_available
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/lib/python3.9/site-packages/nose/case.py", line 197, in runTest
    self.test(*self.arg)
  File "/build/python-locket/src/locket.py-0.2.1/tests/locket_tests.py", line 25, in run_test
    return func(lock_path)
  File "/build/python-locket/src/locket.py-0.2.1/tests/locket_tests.py", line 145, in lock_file_blocks_until_lock_is_available
    locker_2.acquire()
  File "/build/python-locket/src/locket.py-0.2.1/tests/locket_tests.py", line 274, in acquire
    self._process.stdin_write(b"\n")
  File "/usr/lib/python3.9/site-packages/spur/local.py", line 178, in stdin_write
    self._process_stdin.write(value)
BrokenPipeError: [Errno 32] Broken pipe

======================================================================
ERROR: tests.locket_tests.lock_is_acquired_if_available_before_timeout_expires
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/lib/python3.9/site-packages/nose/case.py", line 197, in runTest
    self.test(*self.arg)
  File "/build/python-locket/src/locket.py-0.2.1/tests/locket_tests.py", line 25, in run_test
    return func(lock_path)
  File "/build/python-locket/src/locket.py-0.2.1/tests/locket_tests.py", line 244, in lock_is_acquired_if_available_before_timeout_expires
    locker_2.acquire()
  File "/build/python-locket/src/locket.py-0.2.1/tests/locket_tests.py", line 274, in acquire
    self._process.stdin_write(b"\n")
  File "/usr/lib/python3.9/site-packages/spur/local.py", line 178, in stdin_write
    self._process_stdin.write(value)
BrokenPipeError: [Errno 32] Broken pipe

======================================================================
ERROR: tests.locket_tests.lock_is_released_if_holding_process_is_brutally_killed
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/lib/python3.9/site-packages/nose/case.py", line 197, in runTest
    self.test(*self.arg)
  File "/build/python-locket/src/locket.py-0.2.1/tests/locket_tests.py", line 25, in run_test
    return func(lock_path)
  File "/build/python-locket/src/locket.py-0.2.1/tests/locket_tests.py", line 174, in lock_is_released_if_holding_process_is_brutally_killed
    locker_2.acquire()
  File "/build/python-locket/src/locket.py-0.2.1/tests/locket_tests.py", line 274, in acquire
    self._process.stdin_write(b"\n")
  File "/usr/lib/python3.9/site-packages/spur/local.py", line 178, in stdin_write
    self._process_stdin.write(value)
BrokenPipeError: [Errno 32] Broken pipe

----------------------------------------------------------------------
Ran 15 tests in 0.620s

FAILED (errors=5)
```
Please tell what kind of information would be useful to debug this.

**Comment by mwilliamson** [comment]
The broken pipe suggests to me that the process is dead, although there could be another explanation. My suggestion would be to try and investigate the behaviour of the subprocesses, and see whether they're behaving as expected or not.

**Comment by ArchangeGabriel** [comment]
How can I do that? I have actually no idea at all what this project is supposed to do, I’m only packaging it for Arch Linux because it is a dependency for partd, which in turn is a dependency for dask, which itself is there for spyder, the package I’m actually interested in. ^^

**Comment by mwilliamson** [comment]
I didn't have any specific in mind. I'd have to play around the code to work out exactly how best to debug it.

You mentioned the failures are with the latest release. Do you not get the issue with previous releases?

**Comment by ArchangeGabriel** [comment]
No, the previous release was so old that we had to disable tests entirely, they were failures all the way. And we thought this repo was ~dead (as noted [here](https://github.com/archlinux/svntogit-community/blob/packages/python-locket/trunk/PKGBUILD#L26)).

So actually the current situation is still a large improvement over the previous one and I’ll likely package as-is, but since you revived your repo and fixed almost all the test for our case, I figured out I would ask about these remaining ones. ;)

**Comment by mwilliamson** [comment]
Not so much dead as not needing to be updated -- I don't think there's been any significant change in several years, so I'm not sure why the latest release would have had any effect on the tests passing.

**Comment by ArchangeGabriel** [comment]
Alright I retried with the previous version and actually they are exactly the same failure with the previous version (with only 13 tests ran instead of 15). I guess we had been a bit fast at assuming it was some kind of incompatibility with Py3.9 because the latest version tested here was 3.4.

**Comment by mwilliamson** [comment]
Is there an easy way of reproducing this error? I tried running the QCOW2 image from the [arch-boxes](https://gitlab.archlinux.org/archlinux/arch-boxes) project, but I couldn't tell what the username/password should be.

**Comment by ArchangeGabriel** [comment]
I’ve never used those, but it says the default user is `arch`. Not sure there is a password for it, but I’ll ask.

After that, when inside an Arch system, downloading the above linked PKGBUILD file and running `makepkg -si` in the same folder should try to build the package and run the test suite. At this point is should be possible to navigate within the source directory and try to run things manually.

**Comment by mwilliamson** [comment]
It looks like the arch-boxes project has been updated to include basic images that can be used directly. I tried cloning the repo and running the tests without `makepkg`, and everything worked fine. I also tried running `makepkg -si`, and it seems to work if locket is already installed, which makes me think that the problem is that the processes that the test spawns doesn't have the right Python path, or something along those lines. Hopefully that gives you more information to allow you to debug further.
